src/app/mytools/sidebar-accordion/sidebar-accordion.component.ts

```diff
@@ -3,38 +3,48 @@ import { DockstoreTool } from 'app/shared/swagger';
 import { Observable } from 'rxjs';
 import { ToolQuery } from '../../shared/tool/tool.query';
 import { OrgToolObject } from '../my-tool/my-tool.component';
+import { KeyValue } from '@angular/common';
+import { MetadataService } from '../../shared/swagger/api/metadata.service';
 
-interface groupEntriesBySource {
+interface GroupEntriesByRegistry {
   groupEntryInfo: OrgToolObject<DockstoreTool>[];
-  sourceControlTitle: string;
+  registryTitle: string;
 }
 
 @Component({
   selector: 'app-sidebar-accordion',
   templateUrl: './sidebar-accordion.component.html',
-  styleUrls: ['./sidebar-accordion.component.scss'],
+  styleUrls: ['./sidebar-accordion.component.scss', '../../shared/styles/my-entry-sidebar.scss'],
 })
 export class SidebarAccordionComponent implements OnInit, OnChanges {
   @Input() openOneAtATime;
   @Input() groupEntriesObject: OrgToolObject<DockstoreTool>[];
   @Input() refreshMessage;
   public toolId$: Observable<number>;
+  private dockerRegistryMap = [];
+  public registryToTools: Map<string, GroupEntriesByRegistry> = new Map<string, GroupEntriesByRegistry>([]);
   activeTab = 0;
 
-  public registryToTools: Map<string, groupEntriesBySource> = new Map<string, groupEntriesBySource>([
-    ['quay.io', { groupEntryInfo: [], sourceControlTitle: 'QUAY.IO' }],
-    ['registry.gitlab.com', { groupEntryInfo: [], sourceControlTitle: 'GITLAB.COM' }],
-    ['github.com', { groupEntryInfo: [], sourceControlTitle: 'GITHUB.COM' }],
-    ['dockstore.org', { groupEntryInfo: [], sourceControlTitle: 'DOCKSTORE.ORG' }],
-    ['registry.hub.docker.com', { groupEntryInfo: [], sourceControlTitle: 'HUB.DOCKER.COM' }],
-    // For additional tools, to be removed once amazon and seven bridges have their own groupings
-    ['additional', { groupEntryInfo: [], sourceControlTitle: 'ADDITIONAL' }],
-    // Amazon and seven bridges commented out for now, to be included at a later time
-    // ["", { groupEntryInfo: [], sourceControlTitle: 'AMAZONECR'} ],
-    // ["", { groupEntryInfo: [], sourceControlTitle: 'SEVENBRIDGES'} ],
-  ]);
+  constructor(private toolQuery: ToolQuery, private metadataService: MetadataService) {
+    this.metadataService.getDockerRegistries().subscribe((map) => {
+      this.dockerRegistryMap = map;
+      this.dockerRegistryMap.forEach((registry) => {
+        // Do not create new keys for amazon and seven bridges, to be put in additional category
+        if (registry.dockerPath !== null && registry.dockerPath !== 'public.ecr.aws')
```

Comment:
I don't think this is handling seven bridges like the comment suggests

Reply:
I don't think you should rely on `dockerPath` being null; some day another registry might get added that also is null.

Can you look at `_enum` instead?